util/util.go

```diff
@@ -501,7 +501,14 @@ func ClusterToObjectsMapper(c client.Client, ro runtime.Object, scheme *runtime.
 
 		list := &unstructured.UnstructuredList{}
 		list.SetGroupVersionKind(gvk)
-		if err := c.List(context.TODO(), list, client.MatchingLabels{clusterv1.ClusterLabelName: cluster.Name}); err != nil {
+		if err := c.List(
+			context.TODO(),
+			list,
+			client.InNamespace(cluster.Namespace),
```

Comment:
Not sure if we actually can just set the namespace. I think the func would currently also work with cluster-wide resources (all usages I could find are just using namespaced resources).

Alternatives:
* introduce a new func for namespaced resources
* Is there a way to detect if a resource is namespaced?